frontend/webapp/hooks/destinations/useDestinationFormData.ts

```diff
@@ -1,137 +1,147 @@
+    return ok;
+  };
 
-        // Check if fieldValue is a JSON string that needs stringifying
-        try {
-          const parsedValue = JSON.parse(fieldValue);
+  const loadFormWithDrawerItem = (drawerItem: DrawerBaseItem) => {
+    const {
+      destinationType: { type },
+      name,
+      exportedSignals,
+      fields,
+    } = drawerItem.item as ActualDestination;
+
+    const updatedData: DestinationInput = {
+      ...INITIAL,
+      type,
+      name,
+      exportedSignals,
+      fields: Object.entries(safeJsonParse(fields, {})).map(([key, value]: [string, string]) => ({ key, value })),
+    };
 
-          if (Array.isArray(parsedValue)) {
-            // If it's an array, stringify it for setting the value
-            fieldValue = parsedValue;
-          }
-        } catch (e) {
-          // If parsing fails, it's not JSON, so we keep it as is
-        }
-
-        return {
-          ...field,
-          value: fieldValue,
-        };
-      });
+    setFormData(updatedData);
+  };
 
-      setDynamicFields(df);
-      setExportedSignals(exportedSignals);
-      setSupportedSignals(destinationType.supportedSignals);
+  const { buildFormDynamicFields } = useConnectDestinationForm();
 
-      initialDynamicFieldsRef.current = df;
-      initialExportedSignalsRef.current = exportedSignals;
-      initialSupportedSignalsRef.current = destinationType.supportedSignals;
-    }
-  }, [destinationFields, destination, memoizedBuildFormDynamicFields]);
+  const t = destinationType || formData.type;
+  const { data: { destinationTypeDetails } = {} } = useQuery<DestinationDetailsResponse>(GET_DESTINATION_TYPE_DETAILS, {
+    variables: { type: t },
+    skip: !t,
+  });
 
-  const cardData = useMemo(() => {
-    if (shouldSkip || !isActualDestination(destination?.item) || !destinationFields) {
-      return [{ title: 'Error', value: 'No destination selected or data missing' }];
-    }
+  useEffect(() => {
+    const { logs, metrics, traces } = supportedSignals || {};
 
-    const { exportedSignals, destinationType, fields } = destination.item;
-    const parsedFields = safeJsonParse<Record<string, string>>(fields, {});
-    const destinationDetails = destinationFields.destinationTypeDetails?.fields;
-    const fieldsData = buildDestinationFieldData(parsedFields, destinationDetails);
+    handleFormChange('exportedSignals', {
+      logs: logs?.supported || false,
+      metrics: metrics?.supported || false,
+      traces: traces?.supported || false,
+    });
+  }, [supportedSignals]);
 
-    return [{ title: 'Destination', value: destinationType.displayName || 'N/A' }, { title: 'Monitors', value: buildMonitorsList(exportedSignals) }, ...fieldsData];
-  }, [shouldSkip, destination, destinationFields]);
+  useEffect(() => {
+    if (destinationTypeDetails) {
+      setDynamicFields(
+        buildFormDynamicFields(destinationTypeDetails.fields).map((field) => {
+          if (!!preLoadedFields) {
```

Comment:
move query and useEffect under the states, add error handling for the query, we now have the notification note component